### DISCUSSION TITLE
Why does osu-lazer use both MongoDB and SqlLite?

### DISCUSSION THREAD

**Discussion by Cootz** [ppy/osu#22497] [Q&A] [answered]
I asked this question while trying to get into osu storage source code and game data handling overall. Why does osu use both MongoDB and SqlLite? Does it bring any performance benefits? Please, answer as a dev

**Comment by mk56-spn** [comment, marked as answer]
> I asked this question while trying to get into osu storage source code and game data handling overall. Why does osu use both MongoDB and SqlLite? Does it bring any performance benefits? Please, answer as a dev

If there is still sq lite stuff around it's either obsoleted or in the process of being so . To my knowledge ( assuming you are only referring to the  client ) realm is the only intended DB going forward

**Reply by bdach** [reply]
This is not entirely true. We also use sqlite to read a cached local snapshot of all metadata of online beatmaps that is downloaded on first game start, to speed up lookups that would otherwise need to be performed online. See https://github.com/ppy/osu/blob/master/osu.Game/Beatmaps/BeatmapUpdaterMetadataLookup.cs.

**Reply by mk56-spn** [reply]
> This is not entirely true. We also use sqlite to read a cached local snapshot of all metadata of online beatmaps that is downloaded on first game start, to speed up lookups that would otherwise need to be performed online. See https://github.com/ppy/osu/blob/master/osu.Game/Beatmaps/BeatmapUpdaterMetadataLookup.cs.

Right. Wasn't aware of this. Good to know

**Reply by peppy** [reply]
Of note, the sqlite database is optional and having the dependency missing will not affect too much (mainly batch import speed).

We'll probably eventually look to move this metadata cache to realm. I agree that having any sqlite dependency is something we want to avoid.
